lib/core-complete.sh

```diff
@@ -2572,7 +2572,7 @@ function ble/complete/source:command/.print-command {
     #   function はクォート除去が実行される。従って、compgen -A command には直
     #   接 COMPV を渡し、compgen -A function には compv_quoted を渡す。
     if [[ $slow_compgen ]]; then
-      shopt -q no_empty_cmd_completion && return 0
+      [[ ! $COMPV ]] && shopt -q no_empty_cmd_completion && return 0
```

Comment:
This affects the existing behavior in Cygwin. Have you checked the performance with Cygwin? The current checking for Cygwin was introduced in commit 5327f5d76, but at that time, I probably wanted to turn off the completion with `no_empty_cmd_completion` also for the cases where the current command name starts with `cy*`, `x86*`, etc. Even though the option name implies the "empty" command completion, the user should be turning on this option to suppress the completion that takes an unrealistically long time.

Maybe we can reconsider the condition, but in that case, we should check the consequence in Cygwin, and we should also move this line under the branch of the if-statement at line 2562 (and also we need to split the branch for `[[ ! $COMPV ]]` and `/proc/sys/fs/binfmt_misc/WSInterop`).

Reply:
Also, are you still working on this? Or you've finished this including the check in Cygwin?

Reply:
I checked it with Cygwin but I could not reproduce performance issue.

If consequence in Cygwin has not problem, my opinion is "return without completion only if $COMPV is empty and slow_compgen is 1" is better.

we might should return without completion in following case.
> I probably wanted to turn off the completion with no_empty_cmd_completion also for the cases where the current command name starts with cy*, x86*, etc.

But I thought it can be bit confusing for user because of original `no_empty_cmd_completion` behavior.
Isn't `ble/util/conditional-sync` provide enough mitigation?

Reply:
Thanks. If you've checked that the new behavior doesn't cause a problem in Cygwin, it's just fine to change the behavior.

> Isn't `ble/util/conditional-sync` provide enough mitigation?

The issue with `ble/util/conditional-sync` is that it still performs the processing in the background, which consumes the computational resources. The processing in Cygwin would be generally costful. With `ble/util/conditional-sync`, the line editor would no longer be blocked, but the CPU use would still become high while inputting command names.